tools/kubectl-ownerref/main.go

```diff
@@ -5,60 +5,175 @@ import (
-	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/client-go/util/homedir"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/cmd/get"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/utils/pointer"
 
 	// Load all auth plugins
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 )
 
-var (
-	ns = flag.String("ns", "", "namespace")
-)
-
 func main() {
-	var kubeconfig *string
-	if homeDir := homedir.HomeDir(); homeDir != "" {
-		kubeconfig = flag.String("kubeconfig", filepath.Join(homeDir, ".kube", "config"), "kubeconfig file")
-	} else {
-		kubeconfig = flag.String("kubeconfig", "", "kubeconfig file")
+	klog.InitFlags(nil)
+
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+
+	warningHandler := &warningHandler{
+		fallback: rest.NewWarningWriter(os.Stderr, rest.WarningWriterOptions{Deduplicate: true}),
+	}
+	kubeConfigFlags.WrapConfigFn = func(c *rest.Config) *rest.Config {
+		c.WarningHandler = warningHandler
+		c.QPS = 100
+		return c
 	}
 
-	flag.Parse()
+	matchVersionKubeConfigFlags := cmdutil.NewMatchVersionFlags(kubeConfigFlags)
+
+	f := cmdutil.NewFactory(matchVersionKubeConfigFlags)
+
+	ioStreams := genericclioptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr}
 
-	if err := run(kubeconfig); err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
+	rootCommand := NewCmdGetByOwner("kubectl", f, ioStreams)
+	rootCommand.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	matchVersionKubeConfigFlags.AddFlags(rootCommand.PersistentFlags())
+	kubeConfigFlags.AddFlags(rootCommand.PersistentFlags())
+
+	if err := rootCommand.Execute(); err != nil {
 		os.Exit(1)
 	}
+}
+
+type GetByOwnerOptions struct {
+	PrintFlags *get.PrintFlags
+	ToPrinter  func(*meta.RESTMapping, *bool, bool, bool) (printers.ResourcePrinterFunc, error)
+
+	Namespace         string
+	ExplicitNamespace bool
+
+	AllNamespaces bool
+
+	OwnerKindArg string
+	OwnerKind    schema.GroupKind
+	OwnerName    string
 
+	genericclioptions.IOStreams
 }
 
-func run(kubeconfig *string) error {
-	if len(os.Args) < 3 {
-		fmt.Println("Please complete command: kubectl ownerref [KIND] [NAME]")
-		os.Exit(1)
+type warningHandler struct {
+	fallback rest.WarningHandler
+}
+
+func (h *warningHandler) HandleWarningHeader(code int, agent string, text string) {
+	if code == 299 {
+		// Ignore deprecated kind messages, the user isn't normally specifying the kinds
+		return
+	}
+	h.fallback.HandleWarningHeader(code, agent, text)
+}
+
+// NewGetByOwnerOptions returns a GetByOwnerOptions with default options.
+func NewGetByOwnerOptions(parent string, streams genericclioptions.IOStreams) *GetByOwnerOptions {
+	return &GetByOwnerOptions{
+		PrintFlags: get.NewGetPrintFlags(),
+
+		IOStreams: streams,
+	}
+}
+
+func NewCmdGetByOwner(parent string, f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewGetByOwnerOptions(parent, streams)
+
+	cmd := &cobra.Command{
+		Use: "kubectl ownerref",
```

Comment:
Is it better to add `<KIND> <NAME>` to `Use`?
And, add example refered from [example in README.md](https://github.com/kubernetes-sigs/cluster-addons/blob/master/tools/kubectl-ownerref/README.md#usage) 

```suggestion
		Use: "kubectl ownerref <KIND> <NAME>",
		Example: `kubectl ownerref Dashboard dashboard_sample
kubectl ownerref Dashboard dashboard_sample -o yaml`,
```